gazel/gazel.go

```diff
@@ -621,21 +627,24 @@ func ReconcileRules(pkgPath string, rules []*bzl.Rule, dryRun bool) (bool, error
 		delete(oldRules, r.Name())
 	}
 
-	reconcileLoad(f, rules)
-
 	for _, r := range oldRules {
 		if !RuleIsManaged(r) {
 			continue
 		}
 		f.DelRules(r.Kind(), r.Name())
 	}
+	reconcileLoad(f, f.Rules(""))
+
 	return writeFile(path, f, true, dryRun)
 }
 
 func reconcileLoad(f *bzl.File, rules []*bzl.Rule) {
 	usedRuleKindsMap := map[string]bool{}
 	for _, r := range rules {
-		usedRuleKindsMap[r.Kind()] = true
+		// TODO: make less fragile
+		if strings.HasPrefix(r.Kind(), "go_") || strings.HasPrefix(r.Kind(), "cgo_") {
```

Comment:
Yeah, iterating through all rules in the BUILD file will include non-Go rules which we shouldn't import.

I added a comment.